src/coreComponents/mesh/generators/HexCellBlockManager.cpp

```diff
@@ -0,0 +1,1126 @@
+    CellBlock & cur = group.getGroup< CellBlock >( i );
+    m_cellBlocks[i] = &cur;
+    m_blockCellIndexOffset[i] = prevSum + m_cellBlocks[i]->numElements();
+  }
+  assert( nbBlocks > 0);
+
+  nbElements =  m_blockCellIndexOffset[nbBlocks-1];
+}
+
+// TODO convert to log output
+void MeshConnectivityBuilder::printDebugInformation() const
+{ 
+   std::cout << std::endl
+            << std::endl;
+  
+  std::cout << " Number of blocks : " << m_cellBlocks.size() << std::endl
+            << std::endl;
+  for (unsigned int i = 0; i < m_cellBlocks.size(); ++i)
+  {
+     std::cout << m_cellBlocks[i] << std::endl;
+  }
+  std::cout << std::endl;
+  
+  print(m_blockCellIndexOffset);
+
+  std::cout << " Total number of elements : " << nbElements << std::endl
+            << std::endl;
+
+
+  std::cout << " Number of unique faces : " << nbFaces << std::endl
+            << std::endl;
+
+  std::cout << " Face Info " << std::endl
+            << std::endl;
+
+  print(m_allFacesToNeighbors);
+  std::cout << std::endl
+            << std::endl;
+
+  print(m_uniqueFaces);
+  std::cout << std::endl
+            << std::endl;
+
+  print(m_isBoundaryFace);
+  std::cout << std::endl
+            << std::endl;
+
+  std::cout << " NB Edges : " << nbEdges << std::endl
+            << std::endl;
+
+  std::cout << " Edge Info " << std::endl
+            << std::endl;
+
+  print(m_allEdges);
+  std::cout << std::endl
+          << std::endl;
+  
+  print(m_uniqueEdges);
+  std::cout << std::endl
+            << std::endl;
+
+  return;
+}
+
+std::pair< CellBlockIndex, LocalCellIndex> 
+MeshConnectivityBuilder::getBlockCellFromManagerCell( LocalCellIndex cellId ) const
+{
+  CellBlockIndex b = 0;
+  while( cellId > m_blockCellIndexOffset[b] )
+  {
+    b++;
+  }
+  localIndex offset = b > 0 ? m_blockCellIndexOffset[b-1] : 0;
+  LocalCellIndex c = cellId - offset;
+
+  return std::pair< CellBlockIndex, LocalCellIndex> (b, c);
+}
+
+
+void MeshConnectivityBuilder::computeAllFacesToUniqueFace(
+   std::vector< localIndex > & allFacesToUniqueFace ) const
+{
+  allFacesToUniqueFace.resize( m_allFacesToNeighbors.size(), NO_FACE );
+
+  for( unsigned int curFace = 0; curFace < m_uniqueFaces.size(); ++curFace)
+  {
+    localIndex f = m_uniqueFaces[curFace];
+    allFacesToUniqueFace[f] = curFace;
+    if( !m_isBoundaryFace[curFace] )
+    {
+      localIndex f1 = m_allFacesToNeighbors[f];
+      allFacesToUniqueFace[f1] = curFace;
+    }
+  }  
+}
+
+// Cell type independent
+bool MeshConnectivityBuilder::computeEdgesToNodes ( array2d<localIndex> & edgeToNodes ) const
+{
+  edgeToNodes.resize(nbEdges, 2);
+  // Initialize contents
+  edgeToNodes.setValues< serialPolicy >( NO_ID );
+ 
+  for (unsigned int i = 0; i < m_uniqueEdges.size(); ++i)
+  {
+    EdgeInfo e = m_allEdges[m_uniqueEdges[i]];
+    LocalVertexIndex v0 = e.first / nbNodes;
+    LocalVertexIndex v1 = e.first % nbNodes;
+
+    edgeToNodes[i][0] = v0;
+    edgeToNodes[i][1] = v1;
+  }
+  return true;
```

Comment:
I might changea all functions to void.
The idea is to return false if anything goes wrong with the computation.

Reply:
I'm surely old school but one usually returns `0` in case of success and everything non zero is considered as an error code that can be interpreted: you have a lot of different values to express yourself.

It's an error code, not a success code 😉 
```c++
int err = f();
if ( err ) logError( err );
```

Reply:
I am switching to void